script.js

```diff
@@ -115,25 +95,17 @@
 // Write your solution below.
 // ====================================================================
 
-var testArray = [1, 2, 4, 8, 16, 32, 64, 128];
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
+// var testArray = [1, 2, 4, 8, 16, 32, 64, 128];
+// var randomArray = [];
 
+// // !!! ask, why the numbers come out so weirdly in the console
```

Comment:
If I'm seeing the same output as you are, you're seeing something like [ [1], [4], [128], ... ] right? That's because each element in that output array is an array itself, consisting of a single element. That means that your last line, randomArray.push(randomArrayValue), is actually pushing arrays into the randomArray...